pl_examples/basic_examples/dali_image_classifier.py

```diff
@@ -22,21 +22,21 @@
 from torch.utils.data import random_split
 
 import pytorch_lightning as pl
+from pl_examples import TORCHVISION_AVAILABLE, DALI_AVAILABLE
 
-try:
+if TORCHVISION_AVAILABLE:
     from torchvision.datasets.mnist import MNIST
     from torchvision import transforms
-except Exception:
+else:
     from tests.base.datasets import MNIST
 
-try:
+if DALI_AVAILABLE:
     import nvidia.dali.ops as ops
-    import nvidia.dali.types as types
     from nvidia.dali.pipeline import Pipeline
     from nvidia.dali.plugin.pytorch import DALIClassificationIterator
-except (ImportError, ModuleNotFoundError):
+else:
     warn('NVIDIA DALI is not available')
```

Comment:
I guess this isn't related to this PR, but shouldn't this be a hard crash if DALI isn't available?

Reply:
what do you mean?

Reply:
ok, the case is that with tests running on CPU where we do not install DALI the testing will fail just because of this raising error, an alternative would be to return from this script if DALI is missing without crashing...